head_first_design_patterns/tbruinem/json/src/element.cpp

```diff
@@ -0,0 +1,147 @@
+#include "element.hpp"
+#include <assert.h>
+
+#include "common/iostream.hpp"
+
+namespace json {
+
+const ElementType& Element::Type() const
+{
+	return this->type;
+}
+
+void	Element::SetBasicValues(const ParsedJSON& json) {
+	this->type = json.GetType();
+	this->name = json.GetName();
+	this->value_ = json.GetValue();
+}
```

Comment:
Since you're using raw pointers and ints there is not really any need for move semantics but if you were to switch to not using the raw pointers this is where you would use the move semantics on the parsed json, you could pass it as an rvalue reference `&&` and 'move out' of the object here since this is what's called on the `finalize` function which generally is a nice way to move out of a class